CONTENTdm.js

```diff
@@ -0,0 +1,577 @@
+  attachPDF( doc, item );
+
+  ZU.doGet( xmlUrl.object, function( text ) {
+	// Z.debug(text);                                           // View primary XML doc
+	var scrapeMode;
+	if ( !xmlUrl.object || !text || text.match( "The web server is not configured" ) ) scrapeMode = "Web";
+	else { // Scrape from API XML
+	  var docxml = ( new DOMParser() ).parseFromString( text, "text/xml" );
+	  if ( xmlUrl.item ) { // Scraping from two XML docs
+		ZU.doGet( xmlUrl.item, function( text ) {
+		  var doc2xml = ( new DOMParser() ).parseFromString( text, "text/xml" );
+		  // Z.debug(text);                                   // View secondary XML doc
+		  scrapeToItem( docxml, item, url, docTitle, mapping, "xml", doc2xml );
+		  item.complete();
+		} );
+	  } else { // Scraping from a single XML doc
+		scrapeToItem( docxml, item, url, docTitle, mapping, "xml" );
+		item.complete();
+	  }
+	  return; // Return from the ZU.doGet callback after XML scrape
+	}
+	if ( scrapeMode == "Web" ) {
+	  var headerXPath;
+	  var contentXPath;
+	  var offset;
+	  var xhttp = new XMLHttpRequest();
+	  xhttp.open( "GET", url, false );
+	  xhttp.send();
+	  fullSource = xhttp.responseText;
+
+	  if ( doc.documentElement.innerHTML.match( "CONTENTdm Version" ) ) { // Web scrape for CDM version 6+;
+		scrapeToItem( doc, item, url, docTitle, mapping, "web" );
+	  }
+
+	  else if ( fullSource.match( "CONTENTdm Version 5" ) ) { // Web scrape for CDM Version 5
+		headerXPath = '//table/tbody/tr/td/table/tbody/tr/td/table[2]/tbody/tr[2]/td/table/tbody/tr[2]/td/table/tbody/tr/td[1]';
+		contentXPath = '//table/tbody/tr/td/table/tbody/tr/td/table[2]/tbody/tr[2]/td/table/tbody/tr[2]/td/table/tbody/tr/td[2]';
+		offset = 1;
+		legacyScrapeToItem( doc, item, headerXPath, contentXPath, offset, legacyMapping, docTitle, url );
+	  }
+
+	  else if ( fullSource.match( "CONTENTdm" ) ) { // Web scrape for CDM Version 4 or below
+		headerXPath = '//div/div[3]/table[2]/tbody/tr/td[1]';
+		contentXPath = '//div/div[3]/table[2]/tbody/tr/td[2]';
+		offset = 0;
+		legacyScrapeToItem( doc, item, headerXPath, contentXPath, offset, legacyMapping, docTitle, url );
+	  }
+
+	  item.complete();
+
+	  return; // Return from the ZU.doGet callback after Web scrape
+	}
+  } );
+  return '';
+}
+
+function legacyScrapeToItem( doc, item, headerXPath, contentXPath, offset, mapping, docTitle, url ) {
+  var headers = ZU.xpath( doc, headerXPath ); // Scrapes the whole metadata table and looks for matches after.
+  var contents = ZU.xpath( doc, contentXPath );
+  for ( i = 0; i < contents.length; i++ ) {
+	var header;
+	var content;
+	j = i + offset;
+	if ( headers[ j ].textContent ) {
+	  header = headers[ j ].textContent;
+	  header = camelize( header );
+	}
+	if ( contents[ i ].textContent ) content = contents[ i ].textContent;
+	item[ header ] = content;
+  }
+  for ( var i in mapping ) item[ mapping[ i ][ 0 ] ] = item[ mapping[ i ][ 1 ] ];
```

Comment:
This seems unnecessary, why not assign it correctly in the first place. We also generally use a map for this. So something like `if (map[header]) item[map[header]] = content;`